src/Ipac_Table_Writer/Ipac_Table_Writer.cxx

```diff
@@ -129,38 +191,50 @@ void tablator::Ipac_Table_Writer::write_consecutive_records(const Table &table,
+            Data_Type_Adjuster(table).get_datatypes_for_writing(
+                    Format::Enums::IPAC_TABLE));
+}
+
+/**********************************************************/
 
+void tablator::Ipac_Table_Writer::write_consecutive_records(
+        const Table &table, std::ostream &os, const std::set<size_t> &column_ids,
+        bool is_col_inclusion_list, size_t start_row, size_t num_requested) {
+    Column_List_Handler col_handler(column_ids, is_col_inclusion_list,
+                                    table.columns.size());
     tablator::Ipac_Table_Writer::write_consecutive_records(
-            table, os, start_row, num_requested, datatypes_for_writing);
+            table, os, col_handler.get_included_columns_list(), start_row,
+            num_requested,
+            Data_Type_Adjuster(table).get_datatypes_for_writing(
+                    Format::Enums::IPAC_TABLE));
 }
 
 /**********************************************************/
 
 void tablator::Ipac_Table_Writer::write_selected_records(
         const Table &table, std::ostream &os,
         std::vector<size_t> const &requested_row_ids) {
-    Data_Type_Adjuster adjuster(table);
-    const std::vector<Data_Type> datatypes_for_writing =
-            adjuster.get_datatypes_for_writing(Format::Enums::IPAC_TABLE);
-
-    tablator::Ipac_Table_Writer::write_selected_records(table, os, requested_row_ids,
-                                                        datatypes_for_writing);
+    tablator::Ipac_Table_Writer::write_selected_records(
+            table, os, Column_List_Handler::get_all_column_ids(table.columns.size()),
+            requested_row_ids,
+            Data_Type_Adjuster(table).get_datatypes_for_writing(
+                    Format::Enums::IPAC_TABLE));
 }
 
 /**********************************************************/
 
-void tablator::Ipac_Table_Writer::write_single_record(const Table &table,
-                                                      std::ostream &os, size_t row_id) {
-    Data_Type_Adjuster adjuster(table);
-    const std::vector<Data_Type> datatypes_for_writing =
-            adjuster.get_datatypes_for_writing(Format::Enums::IPAC_TABLE);
-
-    size_t curr_row_offset = row_id * table.row_size();
-    tablator::Ipac_Table_Writer::write_single_record_by_offset(
-            table, os, curr_row_offset, datatypes_for_writing);
+void tablator::Ipac_Table_Writer::write_selected_records(
+        const Table &table, std::ostream &os, const std::set<size_t> &column_ids,
+        bool is_col_inclusion_list, std::vector<size_t> const &requested_row_ids) {
+    Column_List_Handler col_handler(column_ids, is_col_inclusion_list,
+                                    table.columns.size());
+    tablator::Ipac_Table_Writer::write_selected_records(
+            table, os, col_handler.get_included_columns_list(), requested_row_ids,
+            Data_Type_Adjuster(table).get_datatypes_for_writing(
+                    Format::Enums::IPAC_TABLE));
```

Comment:
If `Ipac_Table_Writer` were to store a table reference, it could adjust datatypes while writing, and provide an API for selecting which columns to write out, which would probably reduce boilerplate a bit here. But I feel like there was a good reason not to add state to `Ipac_Table_Writer` that I'm forgetting...